File: algorithms/560. Subarray Sum Equals K/solution.cc
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> presum = { { 0, 1 } };
        int count = 0, sum = 0;
        for ( int n : nums ) {
            sum += n;
            count += presum[sum - k];
            ++presum[sum];
        }
        return count;
    }
};

// 80 / 80 test cases passed.
// Status: Accepted
// Runtime: 39 ms

File: algorithms/560. Subarray Sum Equals K/solution.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        d = {0: 1}
        result = s = 0
        for n in nums:
            s += n
            result += d.get(s - k, 0)
            d[s] = d.get(s, 0) + 1
        return result

# 80 / 80 test cases passed.
# Status: Accepted
# Runtime: 95 ms
